src/framework/sdk/rest-service.ts

```diff
@@ -1,216 +1,229 @@
-                    openingBraceCounter = 0;
-                    closingBraceCounter = 0;
-                }
-                else if (openingBraceCounter === 0 && closingBraceCounter === 0)
-                {
-                    var basicField = input.substring(fieldStartIndex, charIterator - fieldStartIndex - 1).trim();
-                    allFields.push(basicField);
-                    fieldStartIndex = charIterator + 1;
-                }
-            }
-        }
-
-        if (fieldStartIndex < charIterator)
-        {
-            var lastField = input.substring(fieldStartIndex, charIterator - fieldStartIndex).trim();
-            allFields.push(lastField);
-        }
-
-        return allFields;
-    }
-
-    private static buildQueryParams(queryParams: { [key: string]: string }) {
-        let result = "";
-        Object.keys(queryParams).forEach((key) => {
-            const value = queryParams[key];
-            if (value)
-                result += `${key}=${value}&`;
-        });
-
-        if (result !== "") {
-            result = "?" + result;
-            result = result.substring(0, result.length - 1);
-        }
-
-        return result;
-    }
-
-    public static buildItemBaseUrl(itemType: string): string {
-        const serviceUrl = RootUrlService.getServiceUrl();
-        const setName = ServiceMetadata.getSetNameFromType(itemType);
-
-        return `${serviceUrl}${setName}`;
-    }
-}
-
-export class RestSdkTypes {
-    public static readonly Video: string = "Telerik.Sitefinity.Libraries.Model.Video";
-    public static readonly Image: string = "Telerik.Sitefinity.Libraries.Model.Image";
-    public static readonly News: string = "Telerik.Sitefinity.News.Model.NewsItem";
-    public static readonly GenericContent: string = "Telerik.Sitefinity.GenericContent.Model.ContentItem";
-    public static readonly Pages: string = "Telerik.Sitefinity.Pages.Model.PageNode";
-}
+import { CollectionResponse } from "./dto/collection-response";
+import { GenericContentItem } from "./dto/generic-content-item";
+import { SdkItem } from "./dto/sdk-item";
+import { RootUrlService } from "./root-url.service";
+import { ServiceMetadata } from "./service-metadata";
+import { GetAllArgs } from "./services/get-all-args";
+import { ODataFilterSerializer } from "./services/odata-filter-serializer";
+
+export class RestService {
+    public static getItemWithFallback<T extends SdkItem>(itemType: string, id: string, provider: string): Promise<T> {
+        const wholeUrl = `${RestService.buildItemBaseUrl(itemType)}(${id})/Default.GetItemWithFallback()${RestService.buildQueryParams({
+            sf_provider: provider,
+            sf_fallback_prop_names: "*",
+            $select: "*"
+        })}`;
+
+        return fetch(wholeUrl, { headers: { "X-Requested-With": "react" } }).then(x => x.json());
+    }
+
+    public static getTokens(taxonomyUrl: string, queryParamsForMethod: any, contentText: string): any{

```

Comment:
If we make a method get Taxons, we should add all the possible params as a seperate argument. Not just a query string